zeroconf/src/linux/service.rs

```diff
@@ -98,6 +114,7 @@ struct AvahiServiceContext {
     client: Option<Arc<ManagedAvahiClient>>,
     name: Option<CString>,
     kind: CString,
+    subtypes: Option<Vec<CString>>,
```

Comment:
I think it would be better to just represent this without the `Option` and then just have an empty `Vec` if there are no subtypes. I think we can also just add `subtypes` to `AvahiServiceContext#new` instead of adding `with_sub_types`.

Reply:
Makes sense, done.